workers/unity/Packages/com.improbable.gdk.core/Systems/SpatialOSSendSystem.cs

```diff
@@ -7,45 +11,54 @@ namespace Improbable.Gdk.Core
     {
         private WorkerBase worker;
-        private MutableView view;
 
-        private readonly List<int> registeredReplicators = new List<int>();
-        private readonly List<int> commandSenders = new List<int>();
+        private readonly List<SpecifiComponentReplicator> specificComponentReplicators =
+            new List<SpecifiComponentReplicator>();
 
         protected override void OnCreateManager(int capacity)
         {
             base.OnCreateManager(capacity);
 
             worker = WorkerRegistry.GetWorkerForWorld(World);
-            view = worker.View;
 
-            GenerateComponentGroups();
+            GetDefaultComponentReplicators();
         }
 
-        private void GenerateComponentGroups()
+        private void GetDefaultComponentReplicators()
         {
-            foreach (var componentTranslatorPair in view.TranslationUnits)
-            {
-                var componentIndex = componentTranslatorPair.Key;
-                var componentTranslator = componentTranslatorPair.Value;
+            // Find all component specific replicatiors and create an instance.
+            var componentReplicationTypes = AppDomain.CurrentDomain.GetAssemblies()
+                .SelectMany(assembly => assembly.GetTypes())
+                .Where(type => typeof(ComponentReplicationHandler).IsAssignableFrom(type) && !type.IsAbstract);
 
-                var replicationComponentGroup = GetComponentGroup(componentTranslator.ReplicationComponentTypes);
-                componentTranslator.ReplicationComponentGroup = replicationComponentGroup;
+            foreach (var componentReplicationType in componentReplicationTypes)
+            {
+                var componentReplicationHandler =
+                    Activator.CreateInstance(componentReplicationType, new { EntityManager }) as
+                        ComponentReplicationHandler;
 
-                registeredReplicators.Add(componentIndex);
-                commandSenders.Add(componentIndex);
+                specificComponentReplicators.Add(new SpecifiComponentReplicator
+                {
+                    ComponentId = componentReplicationHandler.ComponentId,
+                    Handler = componentReplicationHandler,
+                    ReplicationComponentGroup =
+                        GetComponentGroup(componentReplicationHandler.ReplicationComponentTypes),
+                    CommandReplicationGroups = componentReplicationHandler.CommandTypes
+                        .Select(componentType => GetComponentGroup(componentType)).ToList()
+                });
             }
         }
 
-        public bool TryRegisterCustomReplicationSystem(ComponentType type)
+        public bool TryRegisterCustomReplicationSystem(uint componentId)
         {
-            if (!registeredReplicators.Contains(type.TypeIndex))
+            if (specificComponentReplicators.All(componentReplicator => componentReplicator.ComponentId != componentId))
```

Comment:
If this is not allowed to have any duplicated id's, then why is it a `List` and not a `HashSet` that compares on the ComponentId? (or a `Dictionary`)

Reply:
It iterates over it every frame so want this to be as fast as possible to iterate over, don't mind so much the startup time.